src/jllvm/materialization/ByteCodeCompileLayer.cpp

```diff
@@ -1756,6 +1729,7 @@ void CodeGen::codeGenInstruction(ByteCodeOp operation)
                         // Can throw class loader or linkage related errors.
                         generateEHDispatch();
                     }
+                    classObject->mutateType(referenceType(builder.getContext()));
```

Comment:
I think this should be fixed and redundant after https://github.com/JLLVM/JLLVM/pull/126 lands